### PR TITLE
Migrate body_class, content and date helpers to ES6 

### PR DESCRIPTION
**Pull request by vikaspotluri123**
Refs #9589

- Update code style and migrate body_class, content and date helpers

### CODE REVIEW

File: core/server/helpers/body_class.js [hunk 1]
```diff
@@ -2,48 +2,44 @@
...
 
-    if (_.includes(context, 'paged')) {
+    if (context.includes('paged')) {
         classes.push('paged');
     }
 
-    classes = _.reduce(classes, function (memo, item) {
-        return memo + ' ' + item;
-    }, '');
+    classes = classes.reduce((memo, item) => `${memo} ${item}`, '');
```
Comment (vikaspotluri123) [inline core/server/helpers/body_class.js]: I'm not entirely sure why we don't do `classes.join(' ').trimm()` 🤔

File: core/server/helpers/content.js [hunk 2]
```diff
@@ -6,23 +6,26 @@
...
-    _.keys(truncateOptions).map(function (key) {
-        truncateOptions[key] = parseInt(truncateOptions[key], 10);
-    });
+module.exports = function content(options = {}) {
+    const hash = options.hash || {};
+    const truncateOptions = {};
+    let runTruncate = false;
+
+    for (const key of ['words', 'characters']) {
+        if (hash.hasOwnProperty(key)) {
```
Comment (vikaspotluri123) [inline core/server/helpers/content.js]: I'm not sure it's necessary to use `hasOwnProperty` here 😕

File: core/server/helpers/date.js [hunk 3]
```diff
@@ -3,13 +3,11 @@
...
-    SafeString = proxy.SafeString,
-    i18n = proxy.i18n;
+const {SafeString, i18n} = require('./proxy');
+const moment = require('moment-timezone');
 
 module.exports = function (date, options) {
-    var timezone, format, timeago, timeNow, dateMoment;
+    let timezone;
 
     if (!options && date.hasOwnProperty('hash')) {
```
Comment (vikaspotluri123) [inline core/server/helpers/date.js]: Again, not sure that we need to use `hasOwnProperty` here

File: core/server/helpers/body_class.js (hunk 1 above)
Comment (allouis) [reply core/server/helpers/body_class.js]: No reason not to, and IMO `join` is preferred as it is less cognitive overhead to understand

**Review by allouis — APPROVED** [review]

**Pull request merged**